Source/ACE/Entity/Landblock.cs

```diff
@@ -509,6 +510,25 @@ public void EnqueueBroadcast(Position pos, float distance, params GameMessage[]
         }
 
         /// <summary>
+        /// Enqueues a message for global broadcast, thread safe
+        /// </summary>
+        /// <param name="msg"></param>
+        public void EnqueueGlobalBroadcast(params GameMessage[] msgs)
+        {
+            // Atomically checks and sets the broadcastQueued bit --
+            //    guarantees that if we need a broadcast it will be enqueued in the world-managers broadcast queue exactly once
+            if (Interlocked.CompareExchange(ref broadcastQueued, 1, 0) == 0)
+            {
+                WorldManager.BroadcastQueue.EnqueueAction(new ActionEventDelegate(() => SendGlobalBroadcasts()));
```

Comment:
We have a list of sessions, yes.  It is only changed when people log on/off, and should be ok to lock under normal circumstances (just to grab a copy of the container list).

At a minimum, you need to swap the order of the "enqueue" vs "broadcast" logic.  You wouldn't want to have your thread paused after you triggered the broadcast but before you enqueued.  Ideally, the handler for doing a true server-wide broadcast should just do it and not queue it.